main/src/ca/uwaterloo/flix/language/phase/Regions.scala

```diff
@@ -35,13 +35,16 @@ import scala.collection.immutable.SortedSet
 object Regions {
 
   def run(root: Root)(implicit flix: Flix): Validation[Unit, CompilationMessage] = flix.phase("Regions") {
-    val errs = ParOps.parMap(root.defs)(kv => visitDef(kv._2)).flatten
+    val (recoverable, rest) = ParOps.parMap(root.defs)(kv => visitDef(kv._2)).flatten.partition(_.isInstanceOf[Recoverable])
+    val (unrecoverable, errs) = rest.partition(_.isInstanceOf[Unrecoverable])
```

Comment:
Regions phase can ONLY return RegionVarEscapes which can be soft failure. So I think all this code can be simplified.